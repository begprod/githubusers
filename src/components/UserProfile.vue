<template>
<div class="user-profile">
  <img class="user-profile__avatar" :src="user.avatar_url" />
  <div class="user-profile__info">
    <div class="user-profile__name">
      {{ user.name }}
    </div>
    <div class="user-profile__bio">
      {{ user.bio }}
    </div>
  </div>
</div>
</template>

<script>
export default {
  name: 'UserProfile',
  props: {
    user: {
      type: Object,
      required: true
    }
  }
}
</script>

<style>
  .user-profile__avatar {
    width: 250px;
    height: 250px;
    border-radius: 100%;
  }
  .user-profile__info {
    margin-top: 15px;
  }
  .user-profile__name {
    font-size: 20px;
    font-weight: bold;
  }
</style>
