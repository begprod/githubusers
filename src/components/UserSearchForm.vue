<template>
<form @submit.prevent="$emit('submitted', username)">
  <input type="text" v-model="username">
  <button>Submit</button>
</form>
</template>

<script>
export default {
  name: 'UserSearchForm',
  data () {
    return {
      username: ''
    }
  }
}
</script>
