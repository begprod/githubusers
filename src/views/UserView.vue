<template>
<div>
  <UserSearchForm/>
  <UserProfile :user="user"/>
</div>
</template>

<script>
import UserSearchForm from './UserSearchForm'
import UserProfile from './UserProfile'
export default {
  name: 'UserView',
  components: {
    UserSearchForm,
    UserProfile
  },
  data () {
    return {
      user: {
        name: ''
      }
    }
  }
}
</script>
