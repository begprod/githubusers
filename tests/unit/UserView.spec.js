import { shallowMount } from '@vue/test-utils'
import UserView from '@/views/UserView'
import UserSearchForm from '@/views/UserSearchForm'
import UserProfile from '@/views/UserProfile'

describe('UserView', () => {
  const build = () => {
    const wrapper = shallowMount(UserView, {
      data: () => ({
        user: {}
      })
    })
    return {
      wrapper,
      userSearchForm: () => wrapper.find(UserSearchForm),
      userProfile: () => wrapper.find(UserProfile)
    }
  }
  it('render component', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('render child components', () => {
    const { userSearchForm, userProfile } = build()
    expect(userSearchForm().exists()).toBe(true)
    expect(userProfile().exists()).toBe(true)
  })
  it('user profile props bindings', () => {
    const { wrapper, userProfile } = build()
    wrapper.setData({
      user: {
        name: 'Billy'
      }
    })
    expect(userProfile().vm.user).toBe(wrapper.vm.user)
  })
})
