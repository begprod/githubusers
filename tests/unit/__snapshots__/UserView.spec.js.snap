// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView render component 1`] = `
<div>
  <usersearchform-stub></usersearchform-stub>
  <userprofile-stub user="[object Object]"></userprofile-stub>
</div>
`;
